### PR TITLE
reimplemented the observer pattern

### PR DESCRIPTION
**Pull request by Roller23**
Created a new branch and PR because the previous one was fundamentally wrong

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/SelectEventSubscriber.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.features.DriverFeature;
+import edu.kis.powp.observer.Subscriber;
+
+public class SelectEventSubscriber implements Subscriber {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/SelectEventSubscriber.java]: The name of the subscriber doesn't explain anything at all.
What does the subsriber do?
Optionally: What kind of event are we meant to be subscribed to?

**Review by kowallus — CHANGES_REQUESTED** [review]
OK, but one thing requires attention!

File: src/main/java/edu/kis/powp/jobs2d/drivers/SelectEventSubscriber.java (hunk 1 above)
Comment (Roller23) [reply src/main/java/edu/kis/powp/jobs2d/drivers/SelectEventSubscriber.java]: After a little brainstorming we both agreed that "DriverChangeEventSubscriber" will be a better name for this class

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeEventSubscriber.java [hunk 2]
```diff
@@ -3,7 +3,11 @@
 import edu.kis.powp.jobs2d.features.DriverFeature;
 import edu.kis.powp.observer.Subscriber;
 
-public class SelectEventSubscriber implements Subscriber {
+/**
+ * This class subscribes to the DriverManager's changes (invoked by setCurrentDriver())
+ */
+
+public class DriverChangeEventSubscriber implements Subscriber {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeEventSubscriber.java]: I agree that this subscriber waits for "driver change" but the point of all driver manager subscribers would be to wait for "driver change". The thing that make them differ is question why they are waiting (what they are going to do "about it") This should be at least mentioned (a hint).

**Review by kowallus — CHANGES_REQUESTED** [review]
Almost there

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeEventSubscriber.java (hunk 2 above)
Comment (Roller23) [reply src/main/java/edu/kis/powp/jobs2d/drivers/DriverChangeEventSubscriber.java]: I think the new name (DriverInfoUpdater) encapsulates the class' role better now. We dropped the word Subscriber in favor of readability, but the "Updater" at the end suggests that this class implements the Subscriber interface, so it should be fine

**Pull request merged**